src/quantuminspire/qiskit/backend_qx.py

```diff
@@ -264,6 +270,19 @@ def __validate_no_gates_after_measure(experiment: QasmQobjExperiment) -> None:
                     elif qubit in measured_qubits:
                         raise QisKitBackendError('Operation on qubit {} after measurement'.format(qubit))
 
+        memory_bits = []
+        for instruction in experiment.instructions:
```

Comment:
This is the same loop as above, is it not enough to loop once and check for  

>             1. When qubits are used again after being measured.
>             2. When conditional gate operations (binary controlled gates) are done based on previously measured
>                classical bits

in the same loop?

Reply:
My idea to keep it seperated was that the second block could be deleted when we are also doing non-FSP (another ticket).
I coded now one loop.

Reply:
I see your point, now this whole loop can be bypassed when non-fsp is supported as reuse of qubits after measure will also be allowed. So now, or then, this loop can be extracted into a method that is only called when fsp is used.